go/libraries/doltcore/sqle/dtables/status_ignored_table.go

```
@@ -0,0 +1,196 @@
+	ignorePatterns, err := getIgnorePatterns(ctx, st.rootsProvider)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build a set of unstaged table names for quick lookup
+	unstagedTableNames := buildUnstagedTableNameSet(unstagedTables)
+
+	// Convert status rows to status_ignored rows, adding the ignored column
+	rows := make([]statusIgnoredTableRow, len(statusRows))
+	for i, row := range statusRows {
+		ignored := byte(0)
+		// Only check ignore patterns for unstaged tables
+		if row.isStaged == byte(0) && unstagedTableNames[row.tableName] {
+			ignored = checkIfIgnored(row.tableName, ignorePatterns)
+		}
+		rows[i] = statusIgnoredTableRow{
+			tableName: row.tableName,
+			isStaged:  row.isStaged,
+			status:    row.status,
+			ignored:   ignored,
+		}
+	}
+
+	return &StatusIgnoredItr{rows: rows}, nil
+}
+
+// getIgnorePatterns fetches the ignore patterns from the roots provider.
+func getIgnorePatterns(ctx *sql.Context, rp env.RootsProvider[*sql.Context]) (doltdb.IgnorePatterns, error) {
+	roots, err := rp.GetRoots(ctx)
+	if err != nil {
+		return nil, err
+	}
+	schemas := []string{doltdb.DefaultSchemaName}
+	ignorePatternMap, err := doltdb.GetIgnoredTablePatterns(ctx, roots, schemas)
+	if err != nil {
+		return nil, err
+	}
+	return ignorePatternMap[doltdb.DefaultSchemaName], nil
+}
+
+// buildUnstagedTableNameSet builds a set of unstaged table names for quick lookup.
+func buildUnstagedTableNameSet(unstagedTables []diff.TableDelta) map[string]bool {
+	result := make(map[string]bool, len(unstagedTables))
+	for _, td := range unstagedTables {
+		result[td.CurName()] = true
+	}
+	return result
+}
+
+// checkIfIgnored returns 1 if the table name matches an ignore pattern, 0 otherwise.
+func checkIfIgnored(tableName string, ignorePatterns doltdb.IgnorePatterns) byte {
+	tblNameObj := doltdb.TableName{Name: tableName}
+	result, err := ignorePatterns.IsTableNameIgnored(tblNameObj)
+	if err != nil {
+		return byte(0)
+	}
```

We don't want to silently swallow the error here. I'm not sure it makes sense to outline this function if it's small and  only gets called in one place.